src/js/components/stock-movement-wizard/modals/EditPickModal.jsx

```diff
@@ -245,52 +246,51 @@ class EditPickModal extends Component {
   }
 
   fetchPickPageItem() {
-    const itemsUrl = `/api/stockMovementItems/${this.state.attr.itemId}/details?stepNumber=4`;
-
-    apiClient.get(itemsUrl)
-      .then((resp) => {
-        const pickPageItem = resp.data.data;
-
-        const availableItems = _.map(pickPageItem.availableItems, (avItem) => {
-          // check if this picklist item already exists
-          const picklistItem = _.find(pickPageItem.picklistItems, item => item['inventoryItem.id'] === avItem['inventoryItem.id'] && item['binLocation.id'] === avItem['binLocation.id']);
-
-          if (picklistItem) {
-            return {
-              ...avItem,
-              id: picklistItem.id,
-              quantityPicked: picklistItem.quantityPicked,
-              binLocation: {
-                id: picklistItem['binLocation.id'],
-                name: picklistItem['binLocation.name'],
-                zoneName: picklistItem['binLocation.zoneName'],
-              },
-            };
-          }
+    apiClient.get(STOCK_MOVEMENT_ITEM_BY_ID(this.state.attr.itemId), {
+      params: { stepNumber: 4, showDetails: true },
```

Comment:
> [!NOTE]
> This provides more context for the earlier comment about the @deprecated comment.

So the tricky thing with the "details" nested resource is that it was a compromise we made because we didn't have a good solution to the "multiple representations of a resource" issue described in this ticket (https://pihemr.atlassian.net/browse/OBPIH-2649) and we didn't have time to design all of the resources we needed to implement for the stock movement workflow ahead of time. 

Ignoring the step number layer to this, I didn't want to include the picklist items in the main Stock Movement Items GET endpoint
```
/api/stockMovementItems/:id
```
by default, because then we'd be adding that data every time we made this request.

The "details" nested resource was a way to say "every once in awhile i want you to give me a different view of the stock movement item that includes more details". So it's a completely different representation of the Stock Movement Item. And while it's probably ok to pass that as a request parameter I wanted it to stand out to developers as a separate resource because the JSON response is completely different. I am not sure  

In addition, we can't use a picklist items nested resource here because that would only return the picklist items associated with the stock movement items. For example, let's assume we could 
```
/api/picklists/:id/items?stockMovementItemId=:stockMovementItemId
```
or 
```
/api/picklistItems?stockMovementItemId=:stockMovementItemId
```

Instead what we need is a combination of the two "the lines representing the requested items, along with any allocated items (picklist items associated with them". And to this day I don't really have a good idea about how to solve that problem. And that's why we got step numbers in the Stock Movement API. Ugh.

So while the "details" nested resource might not be the right way to provide a different representation of a resource, it's the current convention and shouldn't be deprecated until we have a better solution. But I'd strongly advocate against a single endpoint returning multiple representations based on request parameters like we're doing with the stepNumber and showDetails for the "read" endpoint along with all of the other requests that handle "stepNumber". 

Using the Content Negotiation (Accepts request header) might be an ... acceptable approach since it's a very explicit thing that the developer needs to do. 
```
 Accepts: application/vnd.org.pih.warehouse.api.stockmovement+json;version=1
 Accepts: application/vnd.org.pih.warehouse.api.stockmovement.details+json;version=2”
```